PyNite/BeamSegZ.py

```diff
@@ -129,14 +130,18 @@ def axial(self, x):
         
         return P1 + (p2 - p1)/(2*L)*x**2 + p1*x
 
-    def Torsion(self):
+    def Torsion(self, x = 0):
         """
         Returns the torsional moment in the segment.
         """
         
         # The torsional moment is constant across the segment
         # This can be updated in the future for distributed torsional forces
-        return self.T1
+
+        if isinstance(x, (int, float)):
```

Comment:
I'm not sure what this change is doing. Can you provide some comments to clarify?

Reply:
The reason for checking whether 'x' is a number or not is that these segment result functions can now be given an 'x' value that is actually an array of x values. The calling function expects the returned value to be an array of length equal to the input array of x values. For most functions, the returned value is calculated as a function of the input 'x' value which means the result is automatically equal in length to to 'x'. As the torsion is not calculated as a function of 'x', we need to manually check if 'x' is an array, and if so, return an array of equal length.

I am going to update these functions with type defintions, as a reminder in future that the function can accept either a number or an array of numbers.

Reply:
Actually scratch that last point. I'm not familiar with type annotations prior to python 3.10, which we would need to support. There is a broader conversation to be had there.